### ISSUE TITLE
Successful Pixel Attack and Threshold Attacks!

### ISSUE THREAD

**Issue by aldahdooh** [Trusted-AI/adversarial-robustness-toolbox#817]
Question?
What is the best parameters to generate high accuracy successful black box attacks using Pixel Attack and Threshold Attacks?

I used the default and non-default parameter values (small and high thresholds), but it doesn't give successful attacks. I used that for mnist and cifar-10 datasets.
```
classifier=model_mnist
kclassifier = KerasClassifier(model=classifier, clip_values=(0, 1))
attack = ThresholdAttack(classifier=kclassifier)
adv_data = attack.generate(x=x_test)
``` 
Is that OK or or there is a bug?

**Comment by beat-buesser** [comment]
Hi @aldahdooh Thank you very much for your interest in ART!

Could you please provide a few more details:
- Does `model_mnist` predict probabilities or logits?
- What is the accuracy achieved by `model_mnist` and can you reproduce the accuracy with `kclassifier.predict`?
- Which version of ART are you using?

@shashankkotyan What do you think?

**Comment by aldahdooh** [comment]
* The model predicts probabilities.
* The acc is 99.32 for *MNIST* and 93.35 for *CIFAR*, and I can reproduce the acc with `kclassifier.predict`.
* I am using ART 1.5.0.

**Comment by shashankkotyan** [comment]
@beat-buesser I think, this issue arises from the same problem mentioned in #384 since both attacks uses CMA-ES as default and `max_iter=1` would have poor performance.

@aldahdooh Please provide for me a few more details. 
- Can you share a minimum working code on Google Colab for us to see?
- Can you pass `es=1` parameter to the attack classes and see if the issue persists?
- Can you provide the stats (either model accuracy or attack accuracy) after the attacks?

**Comment by aldahdooh** [comment]
@shashankkotyan 
@beat-buesser 

Many thanks for your interest.

- Check this [link](https://colab.research.google.com/drive/1yXq1-kfmkcBvsNU05PgkE2M6EvjdScAB?usp=sharing)

- [cnn_mnist.zip](https://github.com/Trusted-AI/adversarial-robustness-toolbox/files/5769365/cnn_mnist.zip)

**Comment by shashankkotyan** [comment]
@beat-buesser This issue arises from the same problem you mentioned in e-mail since both attacks use CMA-ES as default. Therefore, `max_iter=1` in the CMA-ES is causing poor performance.

@aldahdooh This issue was resolved in ART 1.5.1 with the commit 58b349f863e5058f83ab6f11ae14acd7d752c48b.

**Comment by beat-buesser** [comment]
@shashankkotyan Thank you!

@aldahdooh We plan to release ART 1.5.1 by the end of this week. In the meantime you can test the improved attack by installing ART from our branch `dev_1.5.1`
